mixer/adapter/kubernetes2/cache_test.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/istio/mixer/pkg/adapter/test"
+)
+
+func TestEventType_String(t *testing.T) {
+	for e := addition; e <= deletion; e++ {
+		if s := e.String(); s == "Unknown" {
+			t.Errorf("String(%d) => Unknown; want known value", e)
+		}
+	}
+
+	unknown := eventType(4)
+	if unknown.String() != "Unknown" {
+		t.Errorf("String(%d) => %s, want 'Unknown'", unknown, unknown)
+	}
+}
+
+func newfakeInformer() *fakeInformer {
+	f := &fakeInformer{}
+	f.cond = sync.NewCond(f)
+	return f
+}
+
+type fakeInformer struct {
```

Comment:
Rather than creating fake informer, it is better to have fake clientset (i.e. `k8s.io/kubernetes/fake`).